python/py-chainer/Portfile

```diff
@@ -38,5 +40,17 @@ if {${name} ne ${subport}} {
         }
     }
 
+    variant hdf5 description {Support HDF5 serialization} {
+        depends_lib-append      port:py${python.version}-h5py
+    }
+
+    variant imagedataset description {Support image dataset} {
+        depends_lib-append      port:py${python.version}-Pillow
+    }
+
+    variant protobuf description {Caffe model support} {
+        depends_lib-append      port:py${python.version}-protobuf3
+    }
+
     livecheck.type      none
```

Comment:
The problem with just adding dependencies means that if the variant is not selected and the dependency happens to be installed, then it becomes a hidden dependency because the support get compiled in and the port will break if the dependency gets uninstalled. 

So please add the appropriate configure commands to request the support when the variant is selected and to explicitly not add the support if not.

Reply:
@mf2k Thank you for catching that, I would not have noticed it!

Reply:
Thank you for your comment. But I'm afraid I can not understand the situation why this dependency will break the port. Would it be possible to provide an example?

From your explanation, I only can imagine following situation, which will not break py-chainer.
```
sudo port install py27-chainer    # variant is not selected
sudo port install py27-protobuf3    # dependency (py27-protobuf3 is installed)
sudo port uninstall py27-protobuf3    $ dependency gets uninstalled
```

Reply:
You have to switch the order:

`sudo port install py27-protobuf3`
`sudo port install py27-chainer # protobuf variant not selected but support is built because the port happens to be installed`
`sudo port uninstall py27-protobuf3 # Opps. Now it is gone and thus py27-chainer is broken. `